app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description(controller_name, action_name).to_json[:params]
```

Comment:
For 2) I think we should probably trigger `Apipie.app.reload_documentation`, if that method is returning nil for something that clearly exists, it just means the docs have not been loaded. @iNecas is there anything smarter than that we could do?